### PR TITLE
Pages for creating and editing reservations

### LINKED ISSUE #171: Pages for creating and editing reservation

### PR DESCRIPTION
**Pull request by evehmane**
Closes #171

When user clicks timeline slot on search page and goes to page where reservation is created, currently the scroll position remains, which is annoying. However, #174 fixes this.

### CODE REVIEW

**Comment by YoYuUm** [comment]
- [ ] It's missing and error page or redirect when reservation doesn't exist on editing, or resource doesn't exist on create.

**Comment by evehmane** [comment]
Thanks for the good comments!

Missing resource is not handled on creation yet, but I wrote about this to ResourceSelector issue. I think creation page should also allow missing resource and ResourceSelector could render empty resource. So that issue could handle the case.

Instead of loader, now the submit button text changes. I implemented this because it was the fastest solution and already used at least on reservation removal. Also, the UI designer might want some other solution later.

File: app/pages/reservationEdit/ReservationEditPageSelector.js [hunk 1]
```diff
@@ -29,7 +29,13 @@ const resourceSelector = createSelector(
   (resources, resourceId) => resources[resourceId]
 );
 
+const isFetchingSelector = createSelector(
+  resourceSelector,
+  resource => !resource
+);
+
```
Comment (YoYuUm) [inline app/pages/reservationEdit/ReservationEditPageSelector.js]: This won't work if the reservation is not found. Which will avoid the "Varausta ei löytynyt." message to appear.

Maybe you could use "reservationsGetIsActiveSelector || resourceGetIsActiveSelector"

Comment (evehmane) [inline app/pages/reservationEdit/ReservationEditPageSelector.js]: Shieeet, I actually used reservationGetIsActiveSelector before and then for some reason did a fixup which removed that. So I didn't A) use my brain and B) test it after the change :D But good that you noticed! It's fixed now and also uses `resourcesGetIsActiveSelector` as you suggested.

Comment (YoYuUm) [inline app/pages/reservationEdit/ReservationEditPageSelector.js]: No problem Shiet like this happens :)

Now it is fine ;)

**Pull request merged**